File: java/builder/exercise/ConfigurationManager.java
package builder.exercise;

public class ConfigurationManager {
    private int graphicsQuality;
    private boolean enableSound;
    private String controlScheme;
    private boolean enableNotifications;
    // ... other properties

    private ConfigurationManager(ConfigurationBuilder builder) {
        this.graphicsQuality = builder.graphicsQuality;
        this.enableSound = builder.enableSound;
        this.controlScheme = builder.controlScheme;
        this.enableNotifications = builder.enableNotifications;
        // ... set other properties
    }

    // ... getters

    // Nested Builder class
    public static class ConfigurationBuilder {
        private int graphicsQuality;
        private boolean enableSound;
        private String controlScheme;
        private boolean enableNotifications;
        // ... other properties

        public ConfigurationBuilder() {
            // Set default values or leave them uninitialized
        }

        public ConfigurationBuilder setGraphicsQuality(int graphicsQuality) {
            this.graphicsQuality = graphicsQuality;
            return this;
        }

        public ConfigurationBuilder setEnableSound(boolean enableSound) {
            this.enableSound = enableSound;
            return this;
        }

        public ConfigurationBuilder setControlScheme(String controlScheme) {
            this.controlScheme = controlScheme;
            return this;
        }

        public ConfigurationBuilder setEnableNotifications(boolean enableNotifications) {
            this.enableNotifications = enableNotifications;
            return this;
        }

        // ... methods for other properties

        public ConfigurationManager build() {
            return new ConfigurationManager(this);
        }
    }
}

File: C#/Builder/exercise/ConfigurationManager.cs
namespace Builder.Exercise;

public class ConfigurationManager
{
    private ConfigurationManager(ConfigurationBuilder builder)
    {
        GraphicsQuality = builder.GraphicsQuality;
        EnableSound = builder.EnableSound;
        ControlScheme = builder.ControlScheme;
        EnableNotifications = builder.EnableNotifications;
    }

    public int GraphicsQuality { get; }
    public bool EnableSound { get; }
    public string? ControlScheme { get; }
    public bool EnableNotifications { get; }

    public sealed class ConfigurationBuilder
    {
        public int GraphicsQuality { get; private set; }
        public bool EnableSound { get; private set; }
        public string? ControlScheme { get; private set; }
        public bool EnableNotifications { get; private set; }

        public ConfigurationBuilder SetGraphicsQuality(int graphicsQuality)
        {
            GraphicsQuality = graphicsQuality;
            return this;
        }

        public ConfigurationBuilder SetEnableSound(bool enableSound)
        {
            EnableSound = enableSound;
            return this;
        }

        public ConfigurationBuilder SetControlScheme(string controlScheme)
        {
            ControlScheme = controlScheme;
            return this;
        }

        public ConfigurationBuilder SetEnableNotifications(bool enableNotifications)
        {
            EnableNotifications = enableNotifications;
            return this;
        }

        public ConfigurationManager Build()
            => new(this);
    }
}
